jbehave-core/src/test/java/org/jbehave/core/embedder/PerformableTreeBehaviour.java

```diff
@@ -106,6 +111,138 @@ public void shouldNotSkipStoryWhenGivenStoryIsPassed() {
+        EmbedderMonitor embedderMonitor = mock(EmbedderMonitor.class);
+        BatchFailures failures = mock(BatchFailures.class);
+        StoryReporter storyReporter = mock(StoryReporter.class);
+        when(configuration.storyReporter(STORY_PATH)).thenReturn(storyReporter);
+        when(configuration.keywords()).thenReturn(new Keywords());
+        PerformableTree performableTree = new PerformableTree();
+        RunContext runContext = performableTree.newRunContext(configuration, candidateSteps,
+                embedderMonitor, new MetaFilter(), failures);
+        performableTree.addStories(runContext, Collections.singletonList(story));
+        performableTree.perform(runContext, story);
+        verify(storyReporter).failed(eq("When step"), argThat(new ArgumentMatcher<Throwable>() {
+            @Override
+            public boolean matches(Object argument) {
+                return argument instanceof UUIDExceptionWrapper &&
+                        ((UUIDExceptionWrapper) argument).getCause() instanceof DependsOnFailedException;
+            }
+        }));
+    }
+
+    @Test
+    public void shouldPerformScenarioWhenDependsOnIdSuccessful() {
+
+        Scenario scenario1 = new Scenario("base scenario title", Meta.createMeta("@scenarioId id",
+                new Keywords()), null, null, Collections.emptyList());
+        Scenario scenario2 = new Scenario("base scenario title", Meta.createMeta("@dependsOnId id",
+                new Keywords()), null, null, Collections.emptyList());
+        List<Scenario> scenarios = new LinkedList<>();
+        scenarios.add(scenario1);
+        scenarios.add(scenario2);
+        Story story = new Story(STORY_PATH, null, null, null, null,
+                scenarios);
+        Configuration configuration = mock(Configuration.class);
+        when(configuration.stepCollector()).thenReturn(new MarkUnmatchedStepsAsPending());
+        when(configuration.storyControls()).thenReturn(new StoryControls());
+        when(configuration.stepsContext()).thenReturn(new StepsContext());
+        StoryLoader storyLoader = mock(StoryLoader.class);
+        configuration.useStoryLoader(storyLoader);
+        List<CandidateSteps> candidateSteps = Collections.emptyList();
+        EmbedderMonitor embedderMonitor = mock(EmbedderMonitor.class);
+        BatchFailures failures = mock(BatchFailures.class);
+        StoryReporter storyReporter = mock(StoryReporter.class);
+        when(configuration.storyReporter(STORY_PATH)).thenReturn(storyReporter);
+        PerformableTree performableTree = new PerformableTree();
+        RunContext runContext = performableTree.newRunContext(configuration, candidateSteps,
+                embedderMonitor, new MetaFilter(), failures);
+        performableTree.addStories(runContext, Collections.singletonList(story));
+        performableTree.perform(runContext, story);
+        verify(storyReporter, never()).failed(any(), any());
```

Comment:
I believe you need to add more tests to cover all various cases, e.g. there is no test for GivenStories